.github/workflows/ci.yml

```diff
@@ -0,0 +1,98 @@
+name: CI
+
+on: [push]
+
+concurrency:
+  group: ${{github.workflow}}-${{github.ref}}
+  cancel-in-progress: true
+
+env:
+  MAINNET_RPC: ${{ secrets.MAINNET_RPC }}
+  SEPOLIA_RPC: ${{ secrets.SEPOLIA_RPC }}
+
+jobs:
+  unit-tests:
+    name: Run Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn --frozen-lockfile --network-concurrency 1
+
+      - name: Create mock files with smock
+        run: yarn smock
+
+      - name: Precompile using 0.8.14 and via-ir=false
+        run: yarn build
+
+      - name: Run tests
+        shell: bash
+        run: yarn test:unit
+
+  integration-tests:
+    name: Run Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: yarn --frozen-lockfile --network-concurrency 1
+
+      - name: Create mock files with smock
+        run: yarn smock
+
+      - name: Precompile using 0.8.14 and via-ir=false
+        run: yarn build
+
+      - name: Run tests
+        run: yarn test:integration
```

Comment:
Are mock files to be used during integration tests?

Reply:
we need to `forge build`, and cannot specify not to build `test/unit/*`, so w/o mock files those test contracts will fail to compile :/